lib/presentation/pages/registration_page/registration_page.dart

```diff
@@ -1,67 +1,154 @@
+  final _formKey = GlobalKey<FormState>();
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      body: Center(
-        child: Column(
-          children: [
-            const SizedBox(height: AppDimensions.d68),
-            const WelcomeText(),
-            const SizedBox(height: AppDimensions.d40),
-            const WelcomeSubtitle(),
-            const SizedBox(height: AppDimensions.d100),
-            Text(
-              context.tr.registration,
-              style: context.tht.displayMedium,
-            ),
-            const SizedBox(height: AppDimensions.d20),
-            CustomTextField(
-              hintText: context.tr.email,
-              controller: '',
-            ),
-            CustomTextField(
-              hintText: context.tr.password,
-              controller: '',
-              hasPassword: true,
-            ),
-            CustomTextField(
-              hintText: context.tr.repeatPassword,
-              controller: '',
-              hasPassword: true,
-            ),
-            const SizedBox(height: AppDimensions.d50),
-            CustomFloatingActionButton(
-              text: context.tr.registration,
-              onPressed: () {},
-            ),
-            Padding(
-              padding: const EdgeInsets.only(top: AppDimensions.d20),
-              child: Text(
-                context.tr.or,
-                style: context.tht.headlineSmall,
-              ),
-            ),
-            TextButton(
-              onPressed: () => context.router.push(const LoginRoute()),
-              child: Text(
-                context.tr.login,
+      body: SafeArea(
+        child: SingleChildScrollView(
+          child: Center(
+            child: Form(
+              key: _formKey,
+              autovalidateMode: AutovalidateMode.always,
+              child: Column(
+                children: [
+                  const SizedBox(height: AppDimensions.d40),
+                  const WelcomeText(),
+                  const SizedBox(height: AppDimensions.d40),
+                  const WelcomeSubtitle(),
+                  const SizedBox(height: AppDimensions.d80),
+                  Text(
+                    context.tr.registration,
+                    style: context.tht.displayMedium,
+                  ),
+                  const SizedBox(height: AppDimensions.d20),
+                  CustomTextField(
+                    hintText: context.tr.email,
+                    controller: emailController,
+                    validator: (String? value) {
+                      if (value == null || value.isEmpty) {
+                        return context.tr.registration_page_fieldCanNotBeEmpty;
+                      }
+                      if (!value.isValidEmail) {
+                        return context.tr.registration_page_yourEmailIsIncorrect;
+                      }
+                      return null;
+                    },
```

Comment:
it would be better if you moved it to separate function, you can reuse the first if statement in both validators